core/o11y/src/context.rs

```diff
@@ -0,0 +1,28 @@
+use tracing::Span;
+use tracing_opentelemetry::OpenTelemetrySpanExt;
+
+/// Wraps an actix message with the current Span's context.
+/// This lets us trace execution across several actix Actors.
+#[derive(actix::Message, Debug)]
+#[rtype(result = "<T as actix::Message>::Result")]
+pub struct WithSpanContext<T: actix::Message> {
```

Comment:
If we ever want to make this more generally applicable in the future, we could implement the `actix::Message` trait manually, thus removing the `T: actix::Message` bound here. It is fine as-is for now, though. Maybe we will never get to the point where we need this sort of functionality elsehwere.